components/audio_player/audio_player.cpp

```diff
@@ -0,0 +1,875 @@
+/**
```

Comment:
Why .cpp instead .c？

Reply:
So the symbols and functions are in c context, if you look at the header file. I picked .cpp for the source file because I wanted the compiler to better enforce type safety, basically trying to provide flexibility to people with c and c++ codebases to use the component while also adding type safety within that file.

Does using .cpp mess up some projects that aren't enabling cpp usage? Are there things I should and can do to reduce the unintended impact of using the c++ compiler there?